### PR TITLE
[improve doc] Update source rest api doc

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <ranxiaolong716@gmail.com>

### Motivation

Update source rest api doc, as shown below:

image

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SourcesBase.java [hunk 1]
```diff
@@ -63,20 +61,76 @@ public WorkerService get() {
...
+                            "  Possible Values: [ATLEAST_ONCE, ATMOST_ONCE, EFFECTIVELY_ONCE]  \n" +
+                            "ram  \n" +
+                            "  The RAM (in bytes) that need to be allocated per source instance " +
+                            "  (applicable only to the process and Docker runtimes).  \n" +
+                            "schema-type  \n" +
+                            "  The schema type (either a builtin schema like 'avro', 'json', etc.. or  " +
+                            "  custom Schema class name to be used to encode messages emitted from the source  \n" +
+                            "source-config  \n" +
+                            "  Source config key/values  \n" +
+                            "source-config-file  \n" +
```
Comment (jerrypeng) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SourcesBase.java]: this field doesn't exist

**Comment by wolfstudy** [comment]
Yeah, I mistakenly put `./bin/pulsar-admin source` as `sourceConfig`, here should be `sourceConfig`, thanks @jerrypeng  for pointing it out, will fix.

**Comment by wolfstudy** [comment]
@jerrypeng PTAL again, thanks.

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
@wolfstudy a minor detail.  Can you add the following to every endpoint
```
@ApiResponse(code = 503, message = "Function worker service is now initializing. Please try again later.")
```

**Comment by wolfstudy** [comment]
> ```
> @ApiResponse(code = 503, message = "Function worker service is now initializing. Please try again later.")
> ```

sure, thanks

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by wolfstudy** [comment]
run java8 tests

**Comment by wolfstudy** [comment]
run java8 tests

```
2019-06-16\T\06:07:57.367 [ERROR] Caused by: org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SourcesBase.java [hunk 2]
```diff
@@ -63,20 +61,78 @@ public WorkerService get() {
...
+            @ApiParam(value = "The name of namespace")
+            final @PathParam("namespace") String namespace,
+            @ApiParam(value = "The name of source")
+            final @PathParam("sourceName") String sourceName,
+            final @FormDataParam("data") InputStream uploadedInputStream,
+            final @FormDataParam("data") FormDataContentDisposition fileDetail,
+            final @FormDataParam("url") String functionPkgUrl,
+            @ApiParam(
+                    value = "A JSON value presenting source configuration payload. An example of the expected functions can be found here.  \n" +
+                            "tenant  \n" +
```
Comment (yjshen) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SourcesBase.java]: I think tenant, namespace, and name are already encoded in the Path?

**Comment by jiazhai** [comment]
run java8 tests

**Pull request merged**